### PR TITLE
Add sources integration an interface and implementation

### PR DESCRIPTION
**Pull request by adiabramovitch**
Thanks for the contribution, take a moment to read our contributing and security guidelines:

**Checklist**
- [ ] https://github.com/RHEnVision/provisioning-backend#contributing
- [ ] https://github.com/RedHatInsights/secure-coding-checklist

### CODE REVIEW

File: internal/clients/sources/sources_interface.go [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+import (
+	"context"
+)
+
+var GetSourcesImpl func(ctx context.Context) (SourcesIntegration, error)
+
+type SourcesIntegration interface {
+	GetById(ctx context.Context, id string) (*Source, error)
+	List(ctx context.Context) (*[]Source, error)
```
Comment (ezr-ondrej) [inline internal/clients/sources/sources_interface.go:11]: I'd have `ListSourcesWithResponse` and `ShowSourceWithResponse` methods here, that would copy the method interface from the real thing. Wrapping can be useful, but I'd love to stick with the real thing until we hit the need for wrapping..

**Comment by lzap** [comment]
Nice, so we can mock on this level too.

File: internal/clients/sources/sources_real.go [hunk 2]
```diff
@@ -0,0 +1,12 @@
+package sources
+
+import (
+	"context"
+)
+
+var GetSourcesImpl func(ctx context.Context) (SourcesIntegration, error)
```
[the remark this answers is not in the document]
Comment (ezr-ondrej) [reply internal/clients/sources/sources_real.go:7]: Yeah, probably a bit better name :+1:

File: internal/clients/sources/sources_impl.go [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+	"github.com/RHEnVision/provisioning-backend/internal/config"
+)
+
+func getSourcesClient(ctx context.Context) (SourcesIntegration, error) {
+	return NewClientWithResponses(*config.SourcesURL)
+}
+
+func init() {
+	GetSourcesClient = getSourcesClient
+}
```
Comment (lzap) [inline internal/clients/sources/sources_impl.go:15]: Ah can you also rename the file to `sources_interface.go`? I have missed that.

File: internal/clients/sources/sources_real.go [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+import (
+	"context"
+)
+
+var GetSourcesClient func(ctx context.Context) (SourcesIntegration, error)
+
+type SourcesIntegration interface {
+	ShowSourceWithResponse(ctx context.Context, id ID, reqEditors ...RequestEditorFn) (*ShowSourceResponse, error)
+	ListSourcesWithResponse(ctx context.Context, params *ListSourcesParams, reqEditors ...RequestEditorFn) (*ListSourcesResponse, error)
+}
```
Comment (lzap) [inline internal/clients/sources/sources_real.go:12]: Sorry this should have been `sources_interface.go`.

File: internal/clients/sources/sources_impl.go (hunk 3 above)
Comment (lzap) [reply internal/clients/sources/sources_impl.go:15]: Correction, this one is actually correct, the other one.

File: internal/clients/sources/sources_real.go (hunk 4 above)
Comment (lzap) [reply internal/clients/sources/sources_real.go:12]: One detail: We use RequestEditorFn interface from the generated code, ideally we should abstract that.

**Review by lzap — APPROVED** [review]
Abstracting RequestEditorFn and *Response structs is probably something we can do later or when needed.

**Pull request merged**